### PR TITLE
[POI] add details card with actions, remove cancel & events, redsign flows

### LINKED ISSUE #2209: [POI] Add Call/Directions buttons to POI Details card
#### Is your feature request related to a problem? Please describe. 
#29 asks for the ability to dial POI's directly from the Skill. Now that the Android client can support a native dialer with `OpenDefaultApp` event this can be achieved.

#### What is the solution you are looking for?
image

image

- [ ] Capture phone info from Foursquare results
- [ ] Add Call Action.Submit button to card, return `OpenDefaultApp` event with value: {"TelUri":"tel:{telephoneNumber}"}
- [ ] Add Directions Action.Submit button to card, return `OpenDefaultApp` event with value: {"MapsUri":"geo:{latitude},{longitude}"}
- [ ] If user says something like "call" or "get directions", it returns the same events as above

### LINKED ISSUE #2211: [POI] Add optional setting to determine if Azure Maps will provide route information
#### Is your feature request related to a problem? Please describe. 
When demoing the POI Skill it becomes cumbersome to have potential routes returned that are not used because ultimately directions are handled by the client application a user is using. We should by default consolidate the dialog to assume routing is handled by the user's client, and if enabled they can see the current steps of potential routes via Azure Maps.

#### What is the solution you are looking for?
- [ ] Update app settings & related configuration logic to check for this setting
- [ ] Consolidate the dialogs like screenshots shared show
- [ ] Only show a single possible route instead of options

### LINKED ISSUE #2213: [POI] Improve LUIS model for greater understanding and scalability
#### Is your feature request related to a problem? Please describe. 
* The language model captures too many utterances now
* It is not scalable enough when new unmatched entities are found
* It does not follow the naming conventions used by other skills

##### Intents
* Remove NAVIGATION_CANCEL_ROUTE
* Rename NAVIGATION_FIND_PARKING to FindParking
* Rename NAVIGATION_FIND_POINTOFINTEREST to FindPointOfInterest
* Rename NAVIGATION_ROUTE_FROM_X_TO_Y to GetDirections

Intents should understand the different between a user saying something like:
"Find me the nearest coffee shop" -> FindPointOfInterest
"Take me to the nearest coffee shop" -> GetDirections

##### Entities
* Rename ADDRESS to Address
* Rename KEYWORD to Keyword
* Rename POI_TYPE to PoiDescription
* Rename ROUTE_TYPE to RouteDescription

* Add phrase lists to be used as Keyword entities, this way when we need to add new brands they can go into the phrase list instead of being individually labeled.
    * Cusine (Italian, Spanish, Japanese, Polish, etc.)
    * Brand (McDonalds, Panera Bread, Lowe's T-Mobile, etc.)
    * Category (airport, bank, book store, museum, train station, etc.)

### LINKED ISSUE #2215: [POI] Remove the CancelRouteDialog and related logic/state data
#### Is your feature request related to a problem? Please describe. 
The POI Skill should not be managing the state data for an active route and instead be treated as a catalog of POIs to call/get directions to.

#### What is the solution you are looking for?
Remove the CancelRouteDialog and related state/model data that is only used in this scenario

### LINKED ISSUE #2243: [POI] Overview card contents should scale to the card so it renders well on all clients
#### What project is affected?
Point of Interest

#### What language is this in?
C#

#### What happens?
The map is centered and small on the card, leaving a lot of white space

image

#### What are the steps to reproduce this issue?
Trigger POI with the simple "find a coffee shop" query

#### What were you expecting to happen?
The map should scale to the width of the card. This works on the Emulator/Web Chat/Teams, but the Android SDK uses less formatting and helps to highlight the gaps in applying appropriate styles/rules to the cards.

I believe we may just need to apply the "size":"stretch" property to the map image but this may need some experimenting.

#### Any screenshots or additional context?

*this will provide feedback to use on the other cards*

### PR DESCRIPTION
**Pull request by xieofxie**
### Purpose
*What is the context of this pull request? Why is it being done?*

Close #2209 Close #2211 Close #2213 Close #2215  Close #2243

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

For #2209,
* PointOfInterestDetails.json is added
* Additional flow is added after pois are found
* RouteDialog is redesigned as a sub dialog of FindParkingDialog & FindPointOfInterestDialog

For #2211, "Start Navigation" and "Get Directions" are used

For #2215, 
* CancelRouteDialog is not used
* GetDirectionsDialog is added to for NAVIGATION_ROUTE_FROM_X_TO_Y (much like FindPointOfInterestDialog)
* Events.ActiveLocation and Events.ActiveRoute are removed

For #2216, GetDirectionsDialog will send event directly when only one poi is found

For #2243 , "size":"stretch" are added to images, yet it does need tests on Android.

Details Card:
f1

In Teams:
t1

Send event directly (if only one location, no prompt):
e

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

* Update tests
* Remove RouteToPointOfInterestByEventTest, CancelRouteSuccessTest
* Add tests GetDirections(ToNearest)Test, RouteToPointOfInterestThen(Call/StartNavigation)Test

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [x] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

**Comment by xieofxie** [comment]
3 more things to clarify:
1. The removal of NAVIGATION_CANCEL_ROUTE will be in #2213  
2. The "Start" button will be added in a new issue
3. The updates of document will be in a new issue
Thanks~

**Review by ryanisgrig — CHANGES_REQUESTED** [review]
I've tested on the Android environment - a couple of bug/recommendations:

- [ ] When a user says/clicks call, it sends the `OpenDefaultApp` event with an empty value. Value should be ` {"TelUri":"tel"{telephonenumber}"}`
image
- [ ] In the case of no known telephone number, do not show the Call button.
- [ ] Remove the "I found a route. Let's start!" text when showing a route.
- [ ] Remove the "here are more detailed directions" message activity when a user confirms they want to start (can be included in the "Start" button new issue).
image
- [ ] Remove the "Here are details." text when showing a single POI.
image
- [ ] Update casing of buttons to "Show directions" and "Start navigation"
image
- [ ] Remove the speak helper text "the first is POI A, the second is POI b, and the last is POI C".

File: skills/src/csharp/pointofinterestskill/pointofinterestskill/Content/PointOfInterestDetails.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "TextBlock",
+          "id": "Hours",
+          "spacing": "None",
+          "color": "default",
+          "text": "{Hours}",
+          "wrap": true
+        },
+        {
+          "type": "TextBlock",
```
Comment (ryanisgrig) [inline skills/src/csharp/pointofinterestskill/pointofinterestskill/Content/PointOfInterestDetails.json]: Don't show the telephone number, user can just say/click call to get it

**Review by ryanisgrig — CHANGES_REQUESTED** [review]

**Comment by xieofxie** [comment]
- [x] When a user says/clicks call, it sends the `OpenDefaultApp` event with an empty value. Value should be ` {"TelUri":"tel"{telephonenumber}"}`
    - It is caused by empty phone number
- [x] In the case of no known telephone number, do not show the Call button.
- [x] Remove the "I found a route. Let's start!" text when showing a route.
- [x] Remove the "here are more detailed directions" message activity when a user confirms they want to start
    - "Start" is added. Overwriting confirm prompt for "No"
- [x] Remove the "Here are details." text when showing a single POI.
- [x] Update casing of buttons to "Show directions" and "Start navigation"
- [x] Remove the speak helper text "the first is POI A, the second is POI b, and the last is POI C". 

Tests are updated for checking card ids

New card:
f0
f1

In Teams:
t0
t1

**Review by ryanisgrig — APPROVED** [review]
I've gone through and removed the "size":"stretch" properties on images, as now cards on Android seem to hit a max amount of space and get cut off.

**Comment by xieofxie** [comment]
@lzc850612 @lauren-mills @darrenj this needs one more approval. Thanks~

**Comment by xieofxie** [comment]
A small fix since it is not merged..

**Review by lauren-mills — APPROVED** [review]

**Pull request merged**